src/main/c/Objects/pyjclass.c

```diff
@@ -134,17 +136,63 @@ static PyObject* pyjclass_add_inner_classes(JNIEnv *env,
     return (PyObject*) topClz;
 }
 
+/*
+ * Populate an attrs dict with all the Java methods and fields for the given
+ * class. This is currently the only way static methods and fields are made
+ * available. There is no known use for non-static fields but for now they are
+ * left in for backwards compatiblity, consider removing them in the future.
+ */
+static PyObject* pyjclass_init_attr(JNIEnv *env, jclass clazz)
+{
+    PyObject *attr = PyDict_New();
+    if (!attr) {
+        return attr;
```

Comment:
Is the error indicator set by PyDict_New() in this probably-will-never-happen case?

Reply:
The documentation isn't clear on this. I looked at other places this is called in jep code and in cpython code and I can't find anywhere else the error indicator is set manually after this fails so I think the assumption is always that it is setting the error message.